crates/containerd-shim-wasm/src/sandbox/cli.rs

```diff
@@ -36,6 +40,47 @@ macro_rules! revision {
     };
 }
 
+/// Main entry point for the shim with OpenTelemetry tracing.
+///
+/// It parses the `OTEL_EXPORTER_OTLP_ENDPOINT` environment variable to determine
+/// if the shim should be started with OpenTelemetry tracing.
+///
+/// If the environment variable is not set, the shim will be started without tracing.
+/// If the environment variable is empty, the shim will be started without tracing.
+#[cfg(feature = "opentelemetry")]
+pub fn shim_main_with_otel<'a, I>(
+    name: &str,
+    version: &str,
+    revision: impl Into<Option<&'a str>>,
+    shim_version: impl Into<Option<&'a str>>,
+    config: Option<Config>,
+) where
+    I: 'static + Instance + Sync + Send,
+    I::Engine: Default,
+{
+    match std::env::var(OTEL_EXPORTER_OTLP_ENDPOINT) {
```

Comment:
There is also `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` that we should look at, which is an alternative to OTEL_EXPORTER_OTLP_ENDPOINT.

Reply:
This is not supported in the opentelemetry crate. ([issue](https://github.com/open-telemetry/opentelemetry-rust/issues/1892) to track)

Reply:
But i guess we still can support it ourselves.